src/sql/azdata.proposed.d.ts

```diff
@@ -1022,7 +1022,7 @@ declare module 'azdata' {
 		 * @param providerId The table designer provider Id.
 		 * @param tableInfo The table information. The object will be passed back to the table designer provider as the unique identifier for the table.
 		 */
-		export function openTableDesigner(providerId: string, tableInfo: TableInfo): Thenable<void>;
+		export function openTableDesigner(providerId: string, tableInfo: TableInfo, serverInfo: ServerInfo): Thenable<void>;
```

Comment:
please do not use server info, as we discussed previously, you can make the new parameter a key-value pair containing telemetry related information. the table designer feature in ADS core is not aware of anything related to connection, so that in the future we can also design tables in a sql project (the offline scenario) .

Reply:
I'll make it an optional parameter. Sending key value pairs in arguments doesn't make sense since the call comes from the extension.

Reply:
the new parameter should be `telemetryProperties: {[key:string]:string}`, extension should decide what information they want to capture. why doesn't this make sense?

Reply:
When there's no extension telemetry, how would an extension know about core telemetry? The only events explicitly sent from an extension should be from the extension telemetry extension

Reply:
I am not following. for table designer, the extension should have all the information about the table, ADS core doesn't know  anything about the table. in the 2 entry points in mssql extension (new table, designer existing table), we should collect the information and pass to openTableDesigner() method. the properties to be sent are defined in the ADO work item.

Reply:
https://github.com/microsoft/azuredatastudio/blob/fbe10c72c6f06a2f1e7922cbcdfc907aa6467d01/src/sql/azdata.proposed.d.ts#L1025
```export function openTableDesigner(providerId: string, tableInfo: TableInfo): Thenable<void>;```

This is the signature from where the extension calls Table Designer. So you're suggesting to turn this to 

```export function openTableDesigner(providerId: string, tableInfo: TableInfo, telemetryProperties: {[key:string]:string}): Thenable<void>;```

which is really bad design IMO. The extension shouldn't have any telemetry data unless explicitly set by extension authors and that's why we have the extension telemetry library. If a future consumer of this API uses this API, wouldn't they expect telemetry results for themselves but when in reality it will only be used in core?

If you want information about the server, I'll either need to change the signature of `TableInfo` or add this as a parameter.

Reply:
we do want some server related information, but we also need something else for example, SQL server table designer will need table type, which is something only mssql extension knows. the ADS core has no knowledge of it.